### ISSUE TITLE
All phone sounds are routed to the earpiece

### ISSUE THREAD

**Issue by M0rtadha** [Euphoria-OS/Bug-tracker#12]
Do you use Xposed framework?No
Your device: Oneplus One
Which ROM version are you using? 05032015
What kernel are you using - included in the ROM or from another source?
The the one from euphoria: 3.4.67-Bacon Euphoria-OS@OnePlus-One#3

Issue:
Occasionally all the phone's sounds (music playing, notification sounds, general system sounds - like the beep turning the volume up, games sounds, etc) are routed to the earpiece speaker instead of the bottom speaker. I don't know what triggers it, for some reason all sounds are sent to the earpiece making all sounds extremely low and hard to hear (especially if phone is in pocket). only rebooting the phone fixes the issue but it comes back randomly again.

What steps will reproduce the problem?
No steps, it just happens. 

What is the expected result? What do you see instead?
The notification supposed to be routed through the bottom speaker, instead as mentioned notifications and all sounds projected through the earpiece. 

Is the result seen after a clean flash too?
Yes

I posted Logcat report here, but there was an error creating my Issue: body is too long. Therefor I hosted the reports in my google Drive, this is the link:

https://drive.google.com/open?id=0Bwhpcrix4Wf-MGs4WmFjVjlsbkE&authuser=0

**Comment by teeseeuu** [comment]
I see this issue on several Roms. I believe it to be hardware related. Seems to be linked to proximity sensor detecting the near state and routing audio through earpiece. Do you use a screen protector or case? Also, does it happen on other Roms for you? Get a sensor monitoring app and see if when in that state, proximity is stuck on near.

**Comment by M0rtadha** [comment]
I didn't have this problem with stock cyongenmod 11s, it all started when I updated to 12s, therefore I tried other ROMs looking for a fix; this issues presented in all lollipop ROMs I tried like Euphoria, cyongenmod 12 nightly, Oxygen and now ParanoidAndroid. 
I'm not using any screen or phone protector, and my phone usage didn't change. 
I'll install now a sensor monitoring app and I'll check it out, hopefully there is a fix for it. finger crossed it's not a hardware issue, as this is the only bug I'm having and it's driving me crazy.

**Comment by teeseeuu** [comment]
Thanks for the extra info The forums are mixed on hardware/software. This leads me to believe the issue is on the interface between the two. Will keep searching and testing.

**Comment by M0rtadha** [comment]
I checked few proximity apps, and all of it shows me the same results which is 0.0000 and it's not changing, even though as I mentioned I'm not using cover or any screen protector. I'll search for Oxygen Modem firmware and install it, as I didn't have any of this issues when I was using oxygen. 
I'll flash it over Euphoria  and report back if anything changed.

**Comment by M0rtadha** [comment]
Reporting back:

I downloaded Full-oxygen-04-04-15-Bacon-modem-flashable.zip from one of the threads in XDA (http://forum.xda-developers.com/oneplus-one/development/radio-updated-modem-radio-files-oxygen-t3072516) and it didn't workout, like other modems I tried in a search for a fix, but after further research I read that flashing the modem using TWRP recovery doesn't really flash all files properly, as mentioned by one of the forum replies, so I extracted the files from it and only flashed NON-HLOS.bin using fastboot method, and the proximity sensor work perfectly now, and it's been two days without having the sound problems.
It's a relief to know that this is not a hardware issue. But to avoid that in the future releases can you implement something or method that check or make sure that the NON-HLOS.bin file is flashed properly so this problem doesn't occur? or do I have to fastboot it every single time I flash Euphoria ?

Thank you teeseeuu for your Proximity check suggestion, I would have never though of it :)

**Comment by M0rtadha** [comment]
Nope,  I got the bug today after 3 days of complete normal behavior. 
The bug report files is attached in the link below :
https://drive.google.com/folder/d/0Bwhpcrix4Wf-NmFoUFdfSTBvQ2M/edit

**Comment by M0rtadha** [comment]
Any new update regarding the issue? 
I've been having also take week without this issue and today it occurred, this is report files form after it happened I uploaded it:
https://drive.google.com/folder/d/0Bwhpcrix4Wf-YzhnakxQQS1OVkU/edit

Thanks
